src/components/intercept/config/electron-config.tsx

```diff
@@ -142,7 +148,12 @@ class ElectronConfig extends React.Component<{
     }
 
     selectApplication = async () => {
-        const pathToApplication = await uploadFile('path');
+        const useNativePicker =
+            platform == 'win' && window.nativeFile.open;
```

Comment:
This platform check should surely be 'mac', no?

I think to be honest, I think it would be better to use this on all platforms, for consistency, and really to move it into `uploadFile` so that all 'path' uploads use this. What do you think?

I'm only just starting to test it out now, is there any reason you've seen not to use it everywhere?

Reply:
Yes, using `mac` was my intention. While trying some stuff I had changed and then accidentally left this.

I also think that this could be helpful for all platforms, also given that the native dialog can do even more, e.g. specify an appropriate title. On Windows defining a `FileFilter` for executables seems also reasonable. 

When I did this, I just didn't look around in what circumstances `uploadFile` is used but first see whether I can fix it for Electron apps on macOS.